### DISCUSSION TITLE
Struggling to find CMakeSettings.json equivalent in VS Code

### DISCUSSION THREAD

**Discussion by kvnandula04** [microsoft/vscode-cmake-tools#3613] [Q&A] [answered]
I'm struggling to find any information on the docs/forums about how exactly I can setup the same configurations I have on Visual Studio for CMake in VS Code. 

From my understanding, it seems _CMakeSettings.json_ is strictly for Visual Studio which I have and it works fine on Visual Studio. But, when I try to open and build the same codebase on VS code, I get a CMake error because the `build root` property from _CMakeSettings.json_ doesn't apply to VS code thus it doesn't find the right build root. 

I have tried to configure a few things with _CMakePresets.json_ but again I couldn't properly translate from CMakeSettings.json to presets.

I'm essentially trying to find the equivalent of this set of configurations in the VS code's CMake Tools extension:

![image](https://github.com/microsoft/vscode-cmake-tools/assets/34941351/230f008c-25e3-4bf8-8579-dc02f8dcbb90)

Any help on this would be much appreciated!

**Comment by gcampbell-msft** [comment, marked as answer]
@kvnandula04 Apologies for the delay in response. However, the officially supported configuration control using cmake is CMakePresets.json!

There are docs here: https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html, as well as docs for this extension here: https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/cmake-presets.md. 

We also have the `Quick Start` experience which can help you create a CMakePresets.json file and allow you to control your configurations.
